main/index.ts

```diff
@@ -49,20 +55,72 @@ function createWindow(): void {
     if (!isProd) mainWindow.webContents.openDevTools()
   })
 
-  mainWindow.on('closed', () => {
-    mainWindow = null
+  mainWindow.on('minimize', (event: Event) => {
+    event.preventDefault()
+    toggleMainWindows()
+  })
+
+  mainWindow.on('close', (event: Event) => {
+    if (isQuiting) return
+
+    event.preventDefault()
+    toggleMainWindows()
   })
 }
 
-app.on('ready', createWindow)
+function updateTray(): void {
+  const contextMenu: Menu = Menu.buildFromTemplate(
+    (isHidden
+      ? [
+        {
+          click: toggleMainWindows,
+          label: 'Show Electra Desktop',
+        },
+      ]
+      : [
+        {
+          click: toggleMainWindows,
+          label: 'Mimize to Tray',
+        },
+      ]
+    ).concat(
+      [
+        {
+          click: async (): Promise<void> => {
+            isQuiting = true
+            rendererProcessSender.send('app:quit')
+            console.info('Closing Electra daemon...')
+            await communication.electraJs.wallet.stopDaemon()
 
-app.on('window-all-closed', async () => {
-  console.info('Closing Electra daemon...')
-  await communication.electraJs.wallet.stopDaemon()
+            if (process.platform !== 'darwin') app.quit()
+          },
+          label: 'Exit',
+        },
+      ]
+    )
+  )
+  tray.setContextMenu(contextMenu)
+}
 
-  if (process.platform !== 'darwin') app.quit()
-})
+function toggleMainWindows(): void {
+  if (isHidden) {
+    mainWindow.show()
+    isHidden = false
+    updateTray()
+
+    return
+  }
+
+  mainWindow.hide()
+  isHidden = true
+  updateTray()
+}
+
+app.once('ready', () => {
+  tray = new Tray(path.resolve(__dirname, 'assets/logos/128x128.png'))
```

Comment:
![image](https://user-images.githubusercontent.com/36645390/38172209-b8951aec-359f-11e8-83dd-5c337261af41.png)

Icon is too big, even 32x32 is too big for the tray. I think we need a 16x16

Reply:
I repeat it to keep a log of it: despite the bad naming of this branch, this PR concerns only the release preparation for Linux Debian-based os.